### ISSUE TITLE
Graylog is not processing messages

### ISSUE THREAD

**Issue by akrus** [Graylog2/graylog2-server#1699]
Hello!

I've tried googling this issue, but found nothing.

The problem we have is that Graylog status shows the following:
Processing 69 incoming and 0 outgoing msg/s. 897,806 unprocessed messages are currently in the journal, in 6 segments. 69 messages have been appended to, and 0 messages have been read from the journal in the last second.

After graylog-server restart it starts processing messages, does this for few minutes and stops again until next restart:
Processing 46 incoming and 1,146 outgoing msg/s. 897,430 unprocessed messages are currently in the journal, in 6 segments. 45 messages have been appended to, and 0 messages have been read from the journal in the last second.

I've tried playing with configuration, but had no success so far. I've also upgraded Elasticsearch 1.7.2 to 1.7.4, Graylog from 1.2.1 to 1.3.3, changed OpenJDK 7 with Oracle Java 8.

Logs are empty and I cannot find a reason why it doesn't work - we have another instance which works fine (and the configuration is the same). The only difference that comes into my mind is that this one is running under Hyper-V, but I also do not know how this can be related.

I'd really appreciate if someone suggests what we can do there.

**Comment by pdepaepe** [comment]
Probably a ES state issue. What gives you System->Overview page, Elasticsearch cluster section?

Or this call on es node?

curl -XGET localhost:9200/_cluster/health?pretty=true

**Comment by akrus** [comment]
ES is green :(
{
  "cluster_name" : "graylog",
  "status" : "green",
  "timed_out" : false,
  "number_of_nodes" : 2,
  "number_of_data_nodes" : 1,
  "active_primary_shards" : 3,
  "active_shards" : 3,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 0,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0
}

**Comment by pdepaepe** [comment]
Did you also tried to recycle deflector? graylog master logs often give a good indication on what's going bad.

**Comment by akrus** [comment]
Cycled deflector, it executed without any errors, but nothing is processed still.

**Comment by kroepke** [comment]
Could you provide logs from your elasticsearch cluster?
It looks like it cannot write any more data.

**Comment by akrus** [comment]
Sure, here is example:
```
[2016-01-15 16:43:22,214][INFO ][node                     ] [Ryder] version[1.7.4], pid[20682], build[0d3159b/2015-12-15T11:25:18Z]
[2016-01-15 16:43:22,215][INFO ][node                     ] [Ryder] initializing ...
[2016-01-15 16:43:22,329][INFO ][plugins                  ] [Ryder] loaded [], sites []
[2016-01-15 16:43:22,387][INFO ][env                      ] [Ryder] using [1] data paths, mounts [[/var (/dev/sda7)]], net usable_space [177.8gb], net total_space [210.1gb], types [ext4]
[2016-01-15 16:43:25,364][INFO ][node                     ] [Ryder] initialized
[2016-01-15 16:43:25,364][INFO ][node                     ] [Ryder] starting ...
[2016-01-15 16:43:25,563][INFO ][transport                ] [Ryder] bound_address {inet[/0.0.0.0:9300]}, publish_address {inet[/192.168.99.99:9300]}
[2016-01-15 16:43:25,579][INFO ][discovery                ] [Ryder] itsec_graylog/wZeubYSPSJ-Z8nCd6eIsZQ
[2016-01-15 16:43:28,609][INFO ][cluster.service          ] [Ryder] new_master [Ryder][wZeubYSPSJ-Z8nCd6eIsZQ][graylog.local][inet[/192.168.99.99:9300]], reason: zen-disco-join (elected_as_master)
[2016-01-15 16:43:28,649][INFO ][http                     ] [Ryder] bound_address {inet[/0.0.0.0:9200]}, publish_address {inet[/192.168.99.99:9200]}
[2016-01-15 16:43:28,649][INFO ][node                     ] [Ryder] started
[2016-01-15 16:43:28,681][INFO ][gateway                  ] [Ryder] recovered [3] indices into cluster_state
[2016-01-15 16:43:29,777][INFO ][cluster.service          ] [Ryder] added {[graylog2-server][Whl6R8s7Tri-QT8k91KFRg][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false},}, reason: zen-disco-receive(join from node[[graylog2-server][Whl6R8s7Tri-QT8k91KFRg][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false}])
[2016-01-15 16:43:34,127][INFO ][cluster.service          ] [Ryder] removed {[graylog2-server][Whl6R8s7Tri-QT8k91KFRg][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false},}, reason: zen-disco-node_failed([graylog2-server][Whl6R8s7Tri-QT8k91KFRg][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false}), reason transport disconnected
[2016-01-15 16:43:49,557][INFO ][cluster.service          ] [Ryder] added {[graylog2-server][wiAc4UuoRleTBRCIRX9A7A][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false},}, reason: zen-disco-receive(join from node[[graylog2-server][wiAc4UuoRleTBRCIRX9A7A][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false}])
[2016-01-15 16:48:46,693][INFO ][cluster.service          ] [Ryder] removed {[graylog2-server][wiAc4UuoRleTBRCIRX9A7A][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false},}, reason: zen-disco-node_failed([graylog2-server][wiAc4UuoRleTBRCIRX9A7A][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false}), reason transport disconnected
[2016-01-15 16:49:01,812][INFO ][cluster.service          ] [Ryder] added {[graylog2-server][UJgKuBvURA-lceWp4gBPvg][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false},}, reason: zen-disco-receive(join from node[[graylog2-server][UJgKuBvURA-lceWp4gBPvg][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false}])
[2016-01-15 16:54:32,843][INFO ][cluster.service          ] [Ryder] removed {[graylog2-server][UJgKuBvURA-lceWp4gBPvg][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false},}, reason: zen-disco-node_failed([graylog2-server][UJgKuBvURA-lceWp4gBPvg][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false}), reason transport disconnected
[2016-01-15 16:54:46,964][INFO ][cluster.service          ] [Ryder] added {[graylog2-server][e2kigi_ERcyVJe3h7AM6YQ][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false},}, reason: zen-disco-receive(join from node[[graylog2-server][e2kigi_ERcyVJe3h7AM6YQ][graylog.local][inet[/192.168.99.99:9350]]{client=true, data=false, master=false}])
```
Those added/removed messages are Graylog-server restarts I was performing to complete inserts.
Btw, after I moved everything to Linux LXC, everything is fine so far.

**Comment by gravis** [comment]
We are seeing the same behaviour with the docker image `graylog2/allinone`.
After starting, some messages are processed, then it stops with no apparent reason.

We're just seeing the error `service JournalReader [FAILED] Failed to read complete buffer for targetOffset 110982779 startPosition 2147469523` in the logs. Purging the volumes does not solve the problem, it's reproduced every time.

**Comment by gravis** [comment]
For the record, while this issue isn't related to docker, our problem seems to come from the storage driver shipped by default on RedHat OSes (devicemapper). As per my discussion with joschi on IRC, we will try other storage drivers.

**Comment by joschi** [comment]
@gravis Thanks for the update!

**Comment by jimm101** [comment]
I have the same problem... and I'm not following the last 3 comments. @gravis didn't mention ulimit, but is the ulimit issue the update to the docs? Am I reading that correctly?

**Comment by gravis** [comment]
No apparently the issue is when devicemapper is used. 

Anyway, I'm not 100% sure of that now, since data is on a mounted volume, so theoretically, the storage driver isn't involved. We are using glusterfs for volumes, and there might be an issue here with locking instead. We haven't found time to confirm that yet.
We've had a discussion with redhat guys, and they told us devicemapper is only an issue when `loop-lvm` is used, and actually devicemapper with `direct-lvm` is the only driver recommended for production.

@jimm101 could you tell us more about your configuration?

**Comment by bernd** [comment]
@gravis @jimm101 This does not look like a Graylog bug. Please open an issue in the [images](https://github.com/Graylog2/graylog2-images) repository if you still have problems.

Thank you!

**Comment by moricard** [comment]
For the record, I have had the same problem using the docker/allinone image and the `aufs` disk driver. I do, however, see Out of Memory errors from the `JournalReader`.

**Comment by ktamas77** [comment]
our graylog just started to produce the same issue. 6 million messages in the journal and accumulating. the server has a relatively low load, everything is green, no errors.

**Comment by joschi** [comment]
@ktamas77 
We are using GitHub issues for tracking bugs in Graylog itself, but this doesn't look like one. Please post this issue to our [public mailing list](https://groups.google.com/forum/#!forum/graylog2) or join the [#graylog channel on freenode IRC](https://webchat.freenode.net/?channels=%23graylog).

Thank you!

**Comment by BarbaricCorgi** [comment]
Please restart the mongod service

**Comment by kempy007** [comment]
I hit similar issue whereby The journal contains # unprocessed messages. 

This was due to creating a new input for GELF_TCP, I had not applied any extractors, but then copied the ones from the GELF_UDP input after a substantial backlog.

from /system/nodes I initiated a graceful shutdown, and currently the number of unprocessed messages is getting smaller, however this still lost the messages. sending new messages now show in search.

Lesson learned, ensure your input has extractors before allowing bulk messages. 

Hope this helps someone!

**Comment by GTownson** [comment]
In my case my virtual machines storage was at about 96% which was causing issues. I deleted some old indices and restarted both Graylog and Elasticsearch, this freed up some memory but didn't fix the issue. I then delete the Graylog log file as sometimes this has fixed the issue (I have no idea why/how). finally I restarted MongoDB as stated above and that fixed the issue. Thanks BarbaricCorgi for your advice.
